packages/promptable/src/providers/OpenAI.ts

```diff
@@ -238,39 +241,35 @@ export class OpenAI
     > = DEFAULT_OPENAI_EMBEDDINGS_CONFIG
   ) {
     if (Array.isArray(textOrTexts)) {
-      return this.embedMany(textOrTexts, options);
+      return await this.embedMany(textOrTexts, options);
     } else {
-      return this.embedOne(textOrTexts, options);
+      return await this.embedOne(textOrTexts, options);
     }
   }
 
-  private embedOne = async (
-    text: string,
-    options: Omit<CreateEmbeddingRequest, "input">
-  ) => {
+  @retry(3)
+  async embedOne(text: string, options: Omit<CreateEmbeddingRequest, "input">) {
     const result = await this.api.createEmbedding({
       ...options,
       input: text.replace(/\n/g, " "),
     });
 
     return result?.data.data[0].embedding;
-  };
+  }
 
-  private embedMany = async (
+  @retry(3)
+  private async embedMany(
     texts: string[],
     options: Omit<CreateEmbeddingRequest, "input">
-  ) => {
-    const batchResults = await Promise.all(
-      texts.map((text) =>
-        this.api.createEmbedding({
-          ...options,
-          input: text.replace(/\n/g, " "),
-        })
```

Comment:
replace this to a call to this.embedOne, since embedOne has the retry decorator, each function call should be retried in case of failure instead of the whole batch